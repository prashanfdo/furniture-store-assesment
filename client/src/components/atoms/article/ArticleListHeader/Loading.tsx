import ContentLoader from "react-content-loader";

const Loading = () => {
  return (
    <ContentLoader
      data-testid="article-list-header-loading"
      speed={2}
      width={"100%"}
      height={45}
      backgroundColor="#fff"
      foregroundColor="#f0f0f0"
    >
      <rect x="0" y="0" rx="5" ry="5" width="100%" height="40" />
    </ContentLoader>
  );
};

export default Loading;
