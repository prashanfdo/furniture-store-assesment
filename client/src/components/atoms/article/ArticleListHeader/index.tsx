type ArticleListHeaderProps = {
  name: string;
  articleCount: number;
};
export default function ArticleListHeader({ name, articleCount }: ArticleListHeaderProps) {
  return (
    <>
      <h1 className="text-3xl grid grid-cols-[auto_auto_1fr] align-middle mt-4 mb-6">
        {name} <span className="ml-1 text-gray-500">({articleCount})</span>
      </h1>
    </>
  );
}
