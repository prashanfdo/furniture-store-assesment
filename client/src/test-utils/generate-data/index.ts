export * from "./article";
export * from "./category";
export * from "./search";
