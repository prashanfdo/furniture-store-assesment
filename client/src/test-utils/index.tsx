import { render, RenderOptions, renderHook, RenderHookOptions, RenderHookResult } from "@testing-library/react";
import fs from "fs";
import React, { FC, ReactElement } from "react";

import user from "@testing-library/user-event";
import { ProvidersForTests } from "providers";

type GraphQlMocks = React.ComponentProps<typeof ProvidersForTests>["mocks"];

type WrapperProps = React.ComponentProps<typeof ProvidersForTests> & {
  children: React.ReactNode;
};
const Wrapper: FC<WrapperProps> = ({ children, mocks, initialRoute }: WrapperProps) => {
  return (
    <ProvidersForTests mocks={mocks} initialRoute={initialRoute}>
      {children}
    </ProvidersForTests>
  );
};
const wrapper =
  (mocks: GraphQlMocks, initialRoute?: string) =>
  ({ children }: { children: React.ReactNode }) =>
    (
      <Wrapper mocks={mocks} initialRoute={initialRoute}>
        {children}
      </Wrapper>
    );

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper"> & { gqlMocks: GraphQlMocks; initialRoute: string },
) => {
  const view = render(ui, { wrapper: wrapper(options?.gqlMocks, options?.initialRoute), ...options });
  const style = document.createElement("style");
  style.innerHTML = fs.readFileSync("src/index.out.css", "utf8");
  document.head.appendChild(style);

  return view;
};

export const customRenderHook = <T, R>(
  callback: (props: T) => R,
  options?: Omit<RenderHookOptions<T>, "wrapper"> & { gqlMocks?: GraphQlMocks; initialRoute?: string },
): RenderHookResult<R, T> => {
  return renderHook(callback, { ...options, wrapper: wrapper(options?.gqlMocks, options?.initialRoute) });
};

export { faker } from "@faker-js/faker";
export * from "@testing-library/react";
export * from "./generate-data";
export { customRender as render };
export { customRenderHook as renderHook };
export { user };
