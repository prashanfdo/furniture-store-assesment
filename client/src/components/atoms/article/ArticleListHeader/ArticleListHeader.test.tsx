import { render } from "test-utils";
import ArticleListHeader from "./index";

describe("ArticleListHeader", () => {
  it("should render", () => {
    const { container } = render(<ArticleListHeader name="Test Heading" articleCount={123456} />);
    expect(container).toMatchSnapshot();
    expect(document.querySelector("h1")).toBeVisible();
    expect(document.querySelector("h1")).toHaveTextContent("Test Heading (123456)");
  });
});
