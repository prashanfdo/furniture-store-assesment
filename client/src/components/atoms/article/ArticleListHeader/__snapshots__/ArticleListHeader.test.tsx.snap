// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleListHeader should render 1`] = `
<div>
  <h1
    class="text-3xl grid grid-cols-[auto_auto_1fr] align-middle mt-4 mb-6"
  >
    Test Heading
     
    <span
      class="ml-1 text-gray-500"
    >
      (
      123456
      )
    </span>
  </h1>
</div>
`;
